### ISSUE TITLE
[bug] Unhandled Promise Rejection

### ISSUE THREAD

**Issue by SpainTrain** [juanfran/gulp-scss-lint#52]
I am happy to do a PR to fix this, but wanted to file the issue in case you know how to fix it off the top of your head.

When you do not have Ruby and scss-lint installed, one would expect the stream to emit an error and cause the task to fail.  However, it appears that an unhandled promise rejection is causing this not to happen.

```
Unhandled rejection Error in plugin 'gulp-scss-lint'
Message:
    You need to have Ruby and scss-lint gem installed
```

I was able to get the task to fail (and get a stack trace) by adding this to my gulpfile

```js
Promise.onPossiblyUnhandledRejection(function(error) {
    throw new gutil.PluginError('Unhandled Exception', error, {showStack: true});
});
```

which provided the error with a stack trace

```
/home/ubuntu/homebuddy-webapp/node_modules/gulp-scss-lint/src/index.js:28
      .then(function() {
       ^
Error in plugin 'gulp-scss-lint'
Message:
    You need to have Ruby and scss-lint gem installed
    (No stack trace)
From previous event:
    at lint (/home/ubuntu/homebuddy-webapp/node_modules/gulp-scss-lint/src/index.js:28:8)
    at Stream.endStream (/home/ubuntu/homebuddy-webapp/node_modules/gulp-scss-lint/src/index.js:59:7)
    at _end (/home/ubuntu/homebuddy-webapp/node_modules/through/index.js:65:9)
    at Stream.stream.end (/home/ubuntu/homebuddy-webapp/node_modules/through/index.js:74:5)
    at DestroyableTransform.onend (/home/ubuntu/homebuddy-webapp/node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:523:10)
    at DestroyableTransform.g (events.js:260:16)
    at emitNone (events.js:72:20)
    at DestroyableTransform.emit (events.js:166:7)
    at /home/ubuntu/homebuddy-webapp/node_modules/vinyl-fs/node_modules/readable-stream/lib/_stream_readable.js:965:16
    at doNTCallback0 (node.js:407:9)
    at process._tickDomainCallback (node.js:377:13)
npm ERR! Test failed.  See above for more details.
```

This was a pretty bad bug for me because it causes the task to pass, but no subsequent tasks to run.  This meant that our CI was passing, even though our tests never ran :open_mouth: (because we run tests after linting).

**Comment by juanfran** [comment]
thanks @SpainTrain but I can't reproduce the error... What bluebird version are you using?

**Comment by SpainTrain** [comment]
My gulpfile and gulp-scss-lint are both using 2.10.2

```
$ npm -v
3.3.5

$ npm ls bluebird
homebuddy-webapp@0.11.2 /home/spainhower/repos/homebuddy-webapp
└── bluebird@2.10.2 
```

I may have left out a key detail that I have `bundleExec` set to true

```js
gulp.task('scsslint', function() {
    return gulp.src('stylesheets/**/*.scss')
        .pipe(scsslint({bundleExec: true}))
        .pipe(scsslint.failReporter());
});
```

I can try to work up a minimal repro.

**Comment by SpainTrain** [comment]
Reproduced it here: https://github.com/SpainTrain/g-s-l-bug-repro/tree/master

You can see that the exit code is zero despite failure here: https://circleci.com/gh/SpainTrain/g-s-l-bug-repro/1

(I also reproed locally with this PoC)

**Comment by SpainTrain** [comment]
Also, just running the test suite without Ruby installed triggers this as well (tests fail due to timeout rather than the actual error).

**Comment by SpainTrain** [comment]
The error emitted here (https://github.com/juanfran/gulp-scss-lint/blob/master/src/index.js#L33) throws the gutil error, which is why there is an unhandled rejection.  You can test this by appending another `.catch` handler to the end of that and printing the error (it's a gutil error).

I was technically able to get the correct behavior by wrapping those lines in a `setTimeout`:

```js
    }, function(e) {
        setTimeout(function() {
            stream.emit('error', new gutil.PluginError(PLUGIN_NAME, e));
            stream.emit('end');
        }, 1);
    });
```

This is not a fix, but hints at the issues.  Gulp does not appear to be getting that emitted error and `stream.emit('error', Error)` rethrows `Error`.

**Comment by juanfran** [comment]
thanks for all the info!!

I think I've fixed it in the last version 0.3.5

**Comment by SpainTrain** [comment]
looks like it, thanks!
